fbfmaproom/fbfmaproom.py

```diff
@@ -72,33 +72,42 @@
 
 APP.layout = fbflayout.app_layout()
 
-def table_columns(obs_config, obs_dataset_key):
-    def bad_year_css(row):
-        r = row['bad_year']
-        if pd.notna(r):
-            if r == "Bad":
-                return "cell-bad-year"
-            else:
-                return "cell-good-year"
-        else:
-            return ""
+
+def table_columns(obs_config, obs_dataset_keys):
 
     obs_dataset_names = {k: v["label"] for k, v in obs_config.items()}
     tcs = OrderedDict()
-    tcs["year_label"] = dict(name="Year", dynamic=None, style=None,
+    tcs["year_label"] = dict(name="Year", style=None,
                              tooltip="The year whose forecast is displayed on the map")
-    tcs["enso_state"] = dict(name="ENSO State", dynamic=None,
+    tcs["enso_state"] = dict(name="ENSO State",
                              tooltip="Displays whether an El Niño, Neutral, or La Niña state occurred during the year",
                              style=lambda row: {'El Niño': 'cell-el-nino',
                                                 'La Niña': 'cell-la-nina',
                                                 'Neutral': 'cell-neutral'}.get(row['enso_state'], ""))
-    tcs["forecast"] = dict(name="Forecast, %", dynamic=None,
+    tcs["forecast"] = dict(name="Forecast, %",
                            tooltip="Displays all the historical flexible forecast for the selected issue month and location",
                            style=lambda row: "cell-severity-" + str(row['severity']) if row['worst_pnep'] == 1 else "")
-    tcs["obs_rank"] = dict(name=f"{obs_dataset_names[obs_dataset_key]} Rank",
-                           style=lambda row: "cell-severity-" + str(row['severity']) if row['worst_obs'] == 1 else "",
-                           tooltip=None, dynamic=None)
-    tcs["bad_year"] = dict(name="Reported Bad Years", dynamic=None,
+
+    def make_obs_column(obs_key):
+        return dict(
+            name=f"{obs_dataset_names[obs_key]} Rank",
+            style=lambda row: "cell-severity-" + str(row['severity']) if row[f'worst_{obs_key}'] == 1 else "",
+            tooltip=None,
+        )
+
+    for obs_key in obs_dataset_keys:
```

Comment:
So is the idea here that we know whether a particular column is an obs column through in-band messaging via the `_rank` suffix in the key? That probably works well enough

Reply:
No, the obs columns are identified by `obs_dataset_keys`. We never scan through the columns and guess which ones are obs.

Reply:
Oh ok I see. That's probably fine. Do you think it would make more sense to somehow attach that to the table_columns data structure, or not worth it?

Reply:
To what end?

Reply:
My rationale was so the extra parameter would not need to be passed, but unnecessary as this likely temporary ultimately (once we eliminate the forecast-obs distinction)